src/serial.rs

```diff
@@ -1801,27 +1885,54 @@ macro_rules! halUart {
         $USARTX:ident: ($usartX:ident, $apbXenr:ident, $rcc_bit:expr, $usartXen:ident, $pclkX:ident),
     )+) => {
         $(
-        impl<PINS> Serial<$USARTX, PINS> {
-            fn config_stop(self, config: config::Config) -> Self {
-                use crate::stm32::uart4::cr2::STOP_A;
-                use self::config::*;
-
-                self.usart.cr2.write(|w| {
-                    w.stop().variant(match config.stopbits {
-                        StopBits::STOP0P5 => STOP_A::STOP1,
-                        StopBits::STOP1 => STOP_A::STOP1,
-                        StopBits::STOP1P5 => STOP_A::STOP2,
-                        StopBits::STOP2 => STOP_A::STOP2,
-                    })
-                });
-                self
+            impl private::Sealed for $USARTX {}
+            impl Instance for $USARTX {
+                fn ptr() -> *const uart_base::RegisterBlock {
+                    $USARTX::ptr() as *const _
+                }
+
+                fn pclk_freq(clocks: &Clocks) -> u32 {
+                    clocks.$pclkX().0
+                }
+
+                unsafe fn enable_clock(rcc: &crate::stm32::rcc::RegisterBlock) {
+                    use crate::bb;
+
+                    bb::set(&rcc.$apbXenr, $rcc_bit);
+                }
+
+                fn set_stopbits(&self, bits: config::StopBits) {
+                    use crate::stm32::uart4::cr2::STOP_A;
+                    use config::StopBits;
+
+                    self
+                        .cr2
+                        .write(|w| w.stop().variant(
+                            match bits {
+                                StopBits::STOP0P5 => STOP_A::STOP1,
+                                StopBits::STOP1 => STOP_A::STOP1,
+                                StopBits::STOP1P5 => STOP_A::STOP2,
+                                StopBits::STOP2 => STOP_A::STOP2,
```

Comment:
Could you add docs saying that `StopBits::STOP0P5` and `StopBits::STOP1P5` aren't supported when using UART and which `StopBits` will be used instead ?

Reply:
Also, could you squash your commits ?